pkg/pillar/zedcloud/deferred.go

```diff
@@ -68,30 +66,60 @@ type SentHandlerFunction func(
 // sentHandler is callback which will be run on successful sent
 // priorityCheckFunctions may be added to send item with matched itemType firstly
 // default function at the end of priorityCheckFunctions added to serve non-priority items
-func CreateDeferredCtx(zedcloudCtx *ZedCloudContext, sentHandler *SentHandlerFunction,
+func CreateDeferredCtx(zedcloudCtx *ZedCloudContext,
+	ps *pubsub.PubSub, agentName string, ctxName string,
+	warningTime time.Duration, errorTime time.Duration,
+	sentHandler *SentHandlerFunction,
 	priorityCheckFunctions ...TypePriorityCheckFunction) *DeferredContext {
 	// Default "accept all" priority
 	priorityCheckFunctions = append(priorityCheckFunctions,
 		func(obj interface{}) bool {
 			return true
 		})
 
-	deferredCtx := DeferredContext{
+	ctx := &DeferredContext{
 		lock:                   &sync.Mutex{},
 		Ticker:                 flextimer.NewRangeTicker(longTime1, longTime2),
```

Comment:
What is the rationale behind using this ticker with such long period?

Reply:
This sounds to me like "ok there can be bugs and queue can stall, so don't let bugs happen and still try to process the queue in 1d", because when you add something to the queue the timer changed to shorter period, see the startTimer() routine.

Reply:
1d is essentially infinite; timer is running but will not not use any noticable amount of CPU.

Reply:
Not clear why it is needed at all.

Reply:
Need to set it to some value. Can't just delete the timer (since that closes the ticker.C channel and bad things happen). Could set it to maxint64 nanoseconds instead of 1 day if we prefer. A comment would be helpful in any case for the next reader who comes around trying to understand this code.

Reply:
Clear. Will add a comment.